mongodb/README.md

```diff
@@ -0,0 +1,290 @@
+```
+
+#### Restoring from a backup
+
+Only follow this procedure if you are certain you need to restore data. This procedure will cause ALL data to be LOST from the time of the backup.
+
+To restore from backup, first identify which backup you want to restore from in the backup list above.
+
+Update the app:
+`acorn update [APP-NAME] --backup-to-restore [BACKUP FILE NAME]`
+
+This will initiate a `restore-from-backup` job which executes `mongorestore`.
+
+#### Configuring retention policy
+If you would like to keep backups for a specific amount of days, you can configure `--backup-retain-days`.
+```shell
+acorn update [APP-NAME] --backup-retain-days 15
+```
+This will keep backups for 15 days.
+
+
+### Entering diagnostic mode
+Sometimes, due to unexpected issues, installations can become corrupted and get stuck in a `CrashLoopBackOff` restart loop. In these situations, it may be necessary to access the containers and perform manual operations to troubleshoot and fix the issues. To ease this task, this acorn provides the diagnostic mode that will deploy all the containers with all probes disabled and override all commands and arguments with `sleep infinity`.
+
+To activate the diagnostic mode, update the acorn app with the following command.
+```shell
+acorn update [APP-NAME] --diagnostic-mode true
+```
+Once the acorn app has been updated, access the containers by executing the following command:
+```shell
+acorn exec -c mongodb-0 [APP-NAME]
+```
+You can access any container by changing `mongodb-0` with desired container name.
+
+### TLS Support
+If you'd like to enable full TLS encryption, you have to create certificates first.
+#### Generate certificates.
+You can create your own CA and certificates by using `openssl`.
+```shell
+$ openssl genrsa -out ca.key 2048
+$ openssl req -x509 -new -nodes -key ca.key -days 10000 -out ca.crt -subj "/CN=mydomain.com"
+$ cat >openssl.cnf <<EOL
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+[req_distinguished_name]
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = $HOSTNAME1
+DNS.1 = $HOSTNAME2
+EOL
+$ openssl genrsa -out mongo.key 2048
+$ openssl req -new -key mongo.key -out mongo.csr -subj "/CN=$HOSTNAME" -config openssl.cnf
+$ openssl x509 -req -in mongo.csr \
+    -CA ca.crt -CAkey ca.key -CAcreateserial \
+    -out mongo.crt -days 3650 -extensions v3_req -extfile openssl.cnf
+$ cat mongo.crt mongo.key > mongo.pem
+```
+Please replace  `$HOSTNAME` with your actual hostname and `$HOSTNAME1` and `$HOSTNAME2` with alternative hostnames you want to allow access to the MongoDB replica set. You can add more alternative hostnames if needed.
+
+#### Create Acorn secret
+Once certificates are prepared, create an Acorn secret using the following command.
+```shell
+acorn secret create --data=@cert-key=mongo.pem --data=@ca=ca.crt mongo-tls-cert
+```
```

Comment:
Maybe note here the key names `cert-key` and `ca` must be exact.

Reply:
ack